packages/shared/providers.yaml

```diff
@@ -229,6 +229,26 @@ apollo:
             title: API Key
             description: The API key for your Apollo account
 
+apollo-oauth:
+    display_name: Apollo (OAuth)
+    categories:
+        - marketing
+    auth_mode: OAUTH2
+    authorization_url: https://app.apollo.io
+    token_url: https://app.apollo.io/api/v1/oauth/token
+    authorization_url_fragment: oauth/authorize
```

Comment:
not sure I understand the role of `authorization_url_fragment`. why is `the authorization_url` not https://app.apollo.io/oauth/authorize?

Reply:
cannot `authorization_url` be  `https://app.apollo.io/#/oauth/authorize`?

Reply:
Fragments are not sent to the server. In HTTP requests, anything after the `#` in a URL is treated as a fragment identifier and is not transmitted as part of the request to the server. Only the part before the `#` is sent. Therefore, the URL will always be `https://app.apollo.io`.

Reply:
I am not sure I follow. The fragment is concatenated in oauth.controller.ts (changes above). Why do we need to concatenate in the logic instead of being part of the authoriziation_url?

Reply:
I see the confusion. We concatenate the fragment `#/oauth/authorize` in the logic because fragments are only used by the client and are never transmitted to the server as part of an HTTP request. When configuring the `authorizationUrl`, we use the `origin (host)` and `pathname (path)` separately rather than the full href. This is because the current library `simple-oauth2` [expects](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/93a782948d4ca1b68e686e01fe6d9e819aec2423/types/simple-oauth2/v1/index.d.ts#L6) these two components, not the full href. The fragment is therefore omitted in the pathname, which is why I made the above change. 
![image](https://github.com/user-attachments/assets/1ac63383-1a75-40ad-8d0d-f09204b12bd2)

Reply:
We concatenate the fragment `#/oauth/authorize` in the logic because fragments are only used by the client and are never transmitted to the server as part of an HTTP request. When configuring the `authorizationUrl`, we use the origin (host) and pathname (path) separately rather than the full href. This is because the current library (simple-oauth2) expects these two components, not the full href. The fragment is therefore omitted in the pathname, which explains the reasoning behind this approach. Please let me know if I am missing anything here.
![image](https://github.com/user-attachments/assets/57d2462a-c294-453a-92ae-14306ad291e4)

Reply:
got it. It is a shame to have to support fragments even though it is explicitly stated in the oauth2 framework that authorize url should not have fragments: https://www.rfc-editor.org/rfc/rfc6749#section-3.1

Reply:
Yes, I think as we add support for more of these, we are more likely to encounter this kind of situation